newIDE/app/src/ProjectCreation/CreateProjectDialog.js

```diff
@@ -85,13 +102,15 @@ export default class CreateProjectDialog extends React.Component<Props, State> {
               onOpen={onOpen}
               onCreate={onCreate}
               onShowExamples={this._showExamples}
+              pathProvider={this.state.outputPath}
```

Comment:
instead of passing a complex "string store", pass something very simple:
the `outputPath`.

Then it seems that your components needs to change it.
So pass `onChangeOutputPath={(outputPath) => setState({ outputPath })}`

Reply:
Oh, thanks! I don't really know react so I searched up on google how to share a state between parent and child and it said to use a store class. Your solution seems way better.

Reply:
People are a bit obsessed with "stores". In practice, the simplest react way to sharing things is to pass a prop with the value, and a prop with a callback to set the value.
You're not sharing actually, you're giving "just enough stuff" for your children component to do its job. No more no less (which is all the beauty/cleanliness for the components approach).